src/librustc_typeck/check/method/suggest.rs

```diff
@@ -743,8 +743,17 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         // We do this to avoid suggesting code that ends up as `T: FooBar`,
                         // instead we suggest `T: Foo + Bar` in that case.
                         let mut has_bounds = false;
+                        let mut impl_trait = false;
                         if let Node::GenericParam(ref param) = hir.get(id) {
-                            has_bounds = !param.bounds.is_empty();
+                            match param.kind {
+                                hir::GenericParamKind::Type { synthetic: Some(_), .. } => {
+                                    impl_trait = true;  // #63706
```

Comment:
Can you leave a comment describing the output message format and why it differs for synthetic params? IMO opening the issue and tracking down the related error message difference is a bit too far to ask people to travel.

Reply:
Done.